File: app/web/src/services/member.service.ts
import request from './request';
import type { ApiResponse, Member, BalanceRecord, ScoreRecord } from '../types';

export const memberService = {
  // 获取会员信息
  getProfile(): Promise<ApiResponse<Member>> {
    return request.get('/member/profile');
  },

  // 更新会员信息
  updateProfile(data: Partial<Member>): Promise<ApiResponse<Member>> {
    return request.put('/member/profile', data);
  },

  // 获取余额记录
  getBalanceRecords(): Promise<ApiResponse<BalanceRecord[]>> {
    return request.get('/member/balance-records');
  },

  // 获取积分记录
  getScoreRecords(): Promise<ApiResponse<ScoreRecord[]>> {
    return request.get('/member/score-records');
  },
};

File: app/ios/ZayumMember/ZayumMember/Services/MemberService.swift
import Foundation

@Observable
class MemberService {
    static let shared = MemberService()
    
    private let apiClient = APIClient.shared
    
    private init() {}
    
    // MARK: - Profile
    
    func getProfile() async throws -> Member {
        return try await apiClient.get(path: APIEndpoints.profile)
    }
    
    func updateProfile(nickname: String? = nil, 
                       avatar: String? = nil, 
                       email: String? = nil, 
                       gender: Gender? = nil, 
                       birthday: String? = nil) async throws -> Member {
        let request = UpdateProfileRequest(
            nickname: nickname,
            avatar: avatar,
            email: email,
            gender: gender,
            birthday: birthday
        )
        return try await apiClient.put(path: APIEndpoints.profile, body: request)
    }
    
    // MARK: - Balance Records
    
    func getBalanceRecords(page: Int = 1, pageSize: Int = 20) async throws -> [BalanceRecord] {
        let path = "\(APIEndpoints.balanceRecords)?page=\(page)&page_size=\(pageSize)"
        return try await apiClient.get(path: path)
    }
    
    // MARK: - Score Records
    
    func getScoreRecords(page: Int = 1, pageSize: Int = 20) async throws -> [ScoreRecord] {
        let path = "\(APIEndpoints.scoreRecords)?page=\(page)&page_size=\(pageSize)"
        return try await apiClient.get(path: path)
    }
}
